### ISSUE TITLE
crio (1.28.2) service does not start on ubuntu 20.04 its

### ISSUE THREAD

**Issue by ehteshaamkazi** [cri-o/cri-o#7612]
### What happened?

Getting below monitor fields translation error.

```
ekazi@eksre02:~$ sudo crio
[sudo] password for ekazi: 
INFO[2023-12-19 16:09:46.395097748Z] Starting CRI-O, version: 1.28.2, git: unknown(clean) 
INFO[2023-12-19 16:09:46.401451136Z] Node configuration value for hugetlb cgroup is true 
INFO[2023-12-19 16:09:46.401472306Z] Node configuration value for pid cgroup is true 
WARN[2023-12-19 16:09:46.401499834Z] node configuration validation for memoryswap cgroup failed: node not configured with memory swap 
INFO[2023-12-19 16:09:46.401509752Z] Node configuration value for memoryswap cgroup is false 
INFO[2023-12-19 16:09:46.401515286Z] Node configuration value for cgroup v2 is false 
INFO[2023-12-19 16:09:46.414916492Z] Node configuration value for systemd CollectMode is true 
INFO[2023-12-19 16:09:46.426866157Z] Node configuration value for systemd AllowedCPUs is true 
INFO[2023-12-19 16:09:46.427994150Z] Not using native diff for overlay, this may cause degraded performance for building images: kernel has CONFIG_OVERLAY_FS_REDIRECT_DIR enabled 
INFO[2023-12-19 16:09:46.428325754Z] Using default capabilities: CAP_CHOWN, CAP_DAC_OVERRIDE, CAP_FSETID, CAP_FOWNER, CAP_SETGID, CAP_SETUID, CAP_SETPCAP, CAP_NET_BIND_SERVICE, CAP_KILL 
INFO[2023-12-19 16:09:46.449525338Z] Checkpoint/restore support disabled          
INFO[2023-12-19 16:09:46.449585701Z] Using seccomp default profile when unspecified: true 
INFO[2023-12-19 16:09:46.449639444Z] Using the internal default seccomp profile   
INFO[2023-12-19 16:09:46.449692112Z] Installing default AppArmor profile: crio-default 
INFO[2023-12-19 16:09:46.491341768Z] No blockio config file specified, blockio not configured 
INFO[2023-12-19 16:09:46.491403215Z] RDT not available in the host system         
FATA[2023-12-19 16:09:46.495092350Z] validating runtime config: monitor fields translation: failed to translate monitor fields for runtime runc: get conmon version: exit status 1 
ekazi@eksre02:~$ 
```

### What did you expect to happen?

The service should start 

### How can we reproduce it (as minimally and precisely as possible)?

Follow commands in this doc upto sudo systemctl enable crio --now
https://devopscube.com/setup-kubernetes-cluster-kubeadm/?unapproved=65342&moderation-hash=e1e2ce9d6f57c4c7d5ae7e33a4f8cdf6#comment-65342

### Anything else we need to know?

Tried installing crio 1.24 but still it fails.

### CRI-O and Kubernetes version

<details>

```console
$ crio --version
ekazi@eksre02:~$ crio --version
crio version 1.28.2
Version:        1.28.2
GitCommit:      unknown
GitCommitDate:  unknown
GitTreeState:   clean
BuildDate:      2023-12-08T17:14:13Z
GoVersion:      go1.19
Compiler:       gc
Platform:       linux/amd64
Linkmode:       dynamic
BuildTags:      
  apparmor
  seccomp
  containers_image_ostree_stub
  exclude_graphdriver_btrfs
  exclude_graphdriver_devicemapper
  containers_image_openpgp
LDFlags:          -s -w -X github.com/cri-o/cri-o/internal/version.buildDate=2023-12-08T17:14:13Z 
SeccompEnabled:   true
AppArmorEnabled:  false
```

```console
$ kubectl version --output=json
ekazi@eksre02:~$ kubectl version --output=json

Command 'kubectl' not found, but can be installed with:

sudo snap install kubectl


```

</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
ekazi@eksre02:~$ cat /etc/os-release 
NAME="Ubuntu"
VERSION="20.04.6 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.6 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
$ uname -a
ekazi@eksre02:~$ uname -a
Linux eksre02 5.4.0-169-generic #187-Ubuntu SMP Thu Nov 23 14:52:28 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux

```

</details>

### Additional environment details (AWS, VirtualBox, physical, etc.)

<details>
VM running on VSphere ESXi
</details>

**Comment by kwilczynski** [comment]
@ehteshaamkazi, thank you for the report.

I assume that both **runc** and **conmon** binaries are installed and working?

**Comment by ehteshaamkazi** [comment]
Not sure about those. 
I have moved ahead by setting crio 1.27 now cannot verify that scenario since everything is working fine now.

**Comment by kwilczynski** [comment]
@ehteshaamkazi, thank you for the follow-up.

I am glad you got it to work with an older release.

I often build small test clusters for troubleshooting using **kubeadm** and usually the latest versions of CRI-O and its dependencies (runc and conmon) on an Ubuntu-based virtual machine, and I haven't noticed any issues.

However, I am usually using Ubuntu **22.04**, rather than **20.04**—do you have any specific reason for using the older LTS release?

That said, I will try to reproduce your problem on 20.04 too.

**Comment by kwilczynski** [comment]
@ehteshaamkazi, I am unable to reproduce the issues you have seen with CRI-O on Ubuntu 20.04, per the following:

- Ubuntu version

```console
root@worker-node01:~# cat /etc/os-release 
NAME="Ubuntu"
VERSION="20.04.6 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.6 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
```

- Installed software

```console
root@worker-node01:~# dpkg -l | grep -E '(cri-o|runc|conmon|kubelet)'
ii  conmon                                100:2.1.2~0                       amd64        OCI container runtime monitor
ii  cri-o                                 1.28.2~0                          amd64        OCI-based Kubernetes Container Runtime Interface implementation.
ii  cri-o-runc                            1.1.9~1                           amd64        Open Container Project - runtime
ii  kubelet                               1.28.2-00                         amd64        Kubernetes Node Agent
```

- Specific software version

```console
root@worker-node01:~# crio --version
crio version 1.28.2
Version:        1.28.2
GitCommit:      unknown
GitCommitDate:  unknown
GitTreeState:   clean
BuildDate:      2023-12-08T17:14:13Z
GoVersion:      go1.19
Compiler:       gc
Platform:       linux/amd64
Linkmode:       dynamic
BuildTags:      
  apparmor
  seccomp
  containers_image_ostree_stub
  exclude_graphdriver_btrfs
  exclude_graphdriver_devicemapper
  containers_image_openpgp
LDFlags:          -s -w -X github.com/cri-o/cri-o/internal/version.buildDate=2023-12-08T17:14:13Z 
SeccompEnabled:   true
AppArmorEnabled:  true

root@worker-node01:~# /usr/lib/cri-o-runc/sbin/runc --version
runc version unknown
spec: 1.0.2-dev
go: go1.19
libseccomp: 2.5.1
root@worker-node01:~# conmon --version
conmon version 2.1.2
commit: 
root@worker-node01:~# 
```

- Running the CRI-O process manually

```console
root@worker-node01:~# crio -l info
INFO[2023-12-21 03:47:23.704716976Z] Starting CRI-O, version: 1.28.2, git: unknown(clean) 
INFO[2023-12-21 03:47:23.708039320Z] Node configuration value for hugetlb cgroup is true 
INFO[2023-12-21 03:47:23.708186719Z] Node configuration value for pid cgroup is true 
WARN[2023-12-21 03:47:23.708320194Z] node configuration validation for memoryswap cgroup failed: node not configured with memory swap 
INFO[2023-12-21 03:47:23.708422119Z] Node configuration value for memoryswap cgroup is false 
INFO[2023-12-21 03:47:23.708507182Z] Node configuration value for cgroup v2 is false 
INFO[2023-12-21 03:47:23.711744402Z] Node configuration value for systemd CollectMode is true 
INFO[2023-12-21 03:47:23.714543178Z] Node configuration value for systemd AllowedCPUs is true 
INFO[2023-12-21 03:47:23.714880890Z] Not using native diff for overlay, this may cause degraded performance for building images: kernel has CONFIG_OVERLAY_FS_REDIRECT_DIR enabled 
INFO[2023-12-21 03:47:23.715539646Z] Using default capabilities: CAP_CHOWN, CAP_DAC_OVERRIDE, CAP_FSETID, CAP_FOWNER, CAP_SETGID, CAP_SETUID, CAP_SETPCAP, CAP_NET_BIND_SERVICE, CAP_KILL 
INFO[2023-12-21 03:47:23.718941382Z] Checkpoint/restore support disabled          
INFO[2023-12-21 03:47:23.719078509Z] Using seccomp default profile when unspecified: true 
INFO[2023-12-21 03:47:23.719178342Z] Using the internal default seccomp profile   
INFO[2023-12-21 03:47:23.719264956Z] Installing default AppArmor profile: crio-default 
INFO[2023-12-21 03:47:23.734054537Z] No blockio config file specified, blockio not configured 
INFO[2023-12-21 03:47:23.734159787Z] RDT not available in the host system         
INFO[2023-12-21 03:47:23.735285089Z] Conmon does support the --sync option        
INFO[2023-12-21 03:47:23.735385351Z] Conmon does support the --log-global-size-max option 
INFO[2023-12-21 03:47:23.751144634Z] Found CNI network k8s-pod-network (type=calico) at /etc/cni/net.d/10-calico.conflist 
INFO[2023-12-21 03:47:23.752174846Z] Found CNI network crio (type=bridge) at /etc/cni/net.d/100-crio-bridge.conflist 
INFO[2023-12-21 03:47:23.753011502Z] Found CNI network loopback (type=loopback) at /etc/cni/net.d/200-loopback.conflist 
INFO[2023-12-21 03:47:23.753093739Z] Updated default CNI network name to k8s-pod-network 
INFO[2023-12-21 03:47:23.753340441Z] Attempting to restore irqbalance config from /etc/sysconfig/orig_irq_banned_cpus 
INFO[2023-12-21 03:47:23.753535691Z] Restore irqbalance config: failed to get current CPU ban list, ignoring 
INFO[2023-12-21 03:47:23.806987143Z] Got pod network &{Name:dashboard-metrics-scraper-5657497c4c-5qsv4 Namespace:kubernetes-dashboard ID:688fc28a3731e1b0158363d8e54132b173dabd1ca65ef9a7c64e5993929b5363 UID:8df186db-3b63-4b3f-ad49-34390a628596 NetNS:/var/run/netns/2fb5cda1-aabf-4164-a134-318b03801b50 Networks:[] RuntimeConfig:map[k8s-pod-network:{IP: MAC: PortMappings:[] Bandwidth:<nil> IpRanges:[] CgroupPath:kubepods-besteffort-pod8df186db_3b63_4b3f_ad49_34390a628596.slice}] Aliases:map[]} 
INFO[2023-12-21 03:47:23.807364678Z] Checking pod kubernetes-dashboard_dashboard-metrics-scraper-5657497c4c-5qsv4 for CNI network k8s-pod-network (type=calico) 
INFO[2023-12-21 03:47:23.807704553Z] Got pod network &{Name:kubernetes-dashboard-78f87ddfc-bkswm Namespace:kubernetes-dashboard ID:1ad2bd7c37fd59fc20772fc2eaa4ccf17dfddbf14df9fa4f196281f43a37e79d UID:7afb4e8c-9e9c-484d-a463-857e6e633788 NetNS:/var/run/netns/01cd9b9f-434f-459e-9a77-8066406f795e Networks:[] RuntimeConfig:map[k8s-pod-network:{IP: MAC: PortMappings:[] Bandwidth:<nil> IpRanges:[] CgroupPath:kubepods-besteffort-pod7afb4e8c_9e9c_484d_a463_857e6e633788.slice}] Aliases:map[]} 
INFO[2023-12-21 03:47:23.807897906Z] Checking pod kubernetes-dashboard_kubernetes-dashboard-78f87ddfc-bkswm for CNI network k8s-pod-network (type=calico) 
INFO[2023-12-21 03:47:23.808173588Z] Registered SIGHUP reload watcher             
INFO[2023-12-21 03:47:23.808188552Z] Starting seccomp notifier watcher            
INFO[2023-12-21 03:47:23.808206444Z] Create NRI interface                         
INFO[2023-12-21 03:47:23.808216584Z] NRI interface is disabled in the configuration. 
^Croot@worker-node01:~# 
```

Everything works, and there aren't any issues.

**Comment by kwilczynski** [comment]
I will close this issue now because I can't reproduce the same issue you have seen. However, if you run into problems again, please provide us with steps to reproduce the issue and verify that packages have been installed correctly and all the dependent binaries, such as runc, conmon, etc., are working.

/close

**Comment by efficacy38** [comment]
I also encountered this problem; the problem is `conmon` binary is not bundled into the crio package after 1.28.4. Following is how to reproduce this.

I use Debian 12 bookworm and `pkg.k8s.io` as my package repo, and conmon exists at `cri-o=1.28.2-4.1`, but after the upgrade to `crio=1.28.3-1.1`, the conmon binary disappears.

I think Debian's package manifest would be better if a Recommend field that includes Conmon were added.
